### PR TITLE
meta.eql(): call the `eql()` function if defined

### PR DESCRIPTION
**Pull request by jedisct1**
Suggested in #8625

### CODE REVIEW

File: lib/std/meta.zig [hunk 1]
```diff
@@ -691,12 +691,13 @@ test "std.meta.TagPayload" {
...
 
 /// Compares two of any type for equality. Containers are compared on a field-by-field basis,
-/// where possible. Pointers are not followed.
+/// where possible, or using the `eql()` function, if defined. Pointers are not followed.
 pub fn eql(a: anytype, b: @TypeOf(a)) bool {
     const T = @TypeOf(a);
 
     switch (@typeInfo(T)) {
         .Struct => |info| {
+            if (comptime std.meta.trait.hasFn("eql")(@TypeOf(a))) return a.eql(b);
```
Comment (LemonBoy) [inline lib/std/meta.zig:700]: This applies to other container types like unions or opaque ptrs, no?

Comment (jedisct1) [reply lib/std/meta.zig]: Oh, right!

 The test has been moved before the `switch`.

Thank you!

**Comment by g-w1** [comment]
See: https://freenode.irclog.whitequark.org/zig/2021-04-26#1619466090-1619466280;

**Comment by jedisct1** [comment]
> See: https://freenode.irclog.whitequark.org/zig/2021-04-26#1619466090-1619466280;

Oh, thanks! And yeah, that makes sense.

Closing, then 😄

**Pull request closed without merging**